### PR TITLE
feat: Wishlist functionality

### PR DESCRIPTION
**Pull request by ozairkhan2810**
Added Wishlist Functionality

- Add to Wishlist
- Remove from Wishlist
- View Wishlist Page

### CODE REVIEW

File: src/components/navbar/Navbar.js [hunk 1]
```diff
@@ -3,8 +3,12 @@ import { Link } from "react-router-dom";
 
 import "./navbar.css";
 import SearchBar from "./SearchBar";
+import { useWishlist } from "../../context/wishlist/wishlist-context";
 
 const Navbar = () => {
+  const { wishlistState } = useWishlist();
+  const { wishlist } = wishlistState;
+  console.log(wishlist);
```
Comment (fahd23) [inline src/components/navbar/Navbar.js]: remove this logger

**Review by fahd23 — APPROVED** [review]
Looks fine

File: src/components/navbar/Navbar.js [hunk 2]
```diff
@@ -18,20 +22,45 @@ const Navbar = () => {
...
         <SearchBar />
         <Link to="/login" className="header-link">
           <div className="header-login-link">Login</div>
         </Link>
         <Link to="/wishlist" className="header-link">
           <div className="header-wishlist-link rel">
-            <span className="material-icons"> favorite_border </span>
-            <span className="badge-icon">5</span>
+            <span className="material-icons">
+              {" "}
```
Comment (JantuDeb) [inline src/components/navbar/Navbar.js]: empty jsx expression

File: src/pages/wishlist/Wishlist.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import "../../components/products/productList.css";
+
+const Wishlist = () => {
+  const { wishlistState, wishlistDispatch } = useWishlist();
+  const { wishlist } = wishlistState;
+
+  useEffect(() => {
+    document.title = "My Wishlist | CricKit";
+  });
+
```
Comment (JantuDeb) [inline src/pages/wishlist/Wishlist.js]: Do this in the initial render. 
```
useEffect(() => {
    document.title = "My Wishlist | CricKit";
  },[]);
```

**Review by JantuDeb — APPROVED** [review]
Overall the code looks good to me

File: src/components/navbar/Navbar.js [hunk 4]
```diff
@@ -18,20 +22,45 @@ const Navbar = () => {
...
+              favorite_border
+              {wishlist && wishlist.length > 0 && (
+                <span
+                  className="badge-ico"
+                  style={{
+                    position: "absolute",
+                    top: "-6px",
+                    fontSize: "1.25rem",
+                  }}
```
Comment (sushil-mod) [inline src/components/navbar/Navbar.js]: if possible you can remove inline style

File: src/components/navbar/Navbar.js [hunk 5]
```diff
@@ -18,20 +22,45 @@ const Navbar = () => {
...
         <Link to="/cart" className="header-link">
           <div className="header-cart-link rel mr-1">
             <span className="material-icons"> shopping_cart </span>
-            <span className="badge-icon">5</span>
+            <span
+              style={{
+                position: "absolute",
+                top: "-10px",
+                fontSize: "1.25rem",
+              }}
```
Comment (sushil-mod) [inline src/components/navbar/Navbar.js]: same here

**Review by sushil-mod — APPROVED** [review]
some suggestion ,rest is good

**Pull request merged**